core/block/editor/dataview/dataview.go

```diff
@@ -112,6 +112,31 @@ func (d *sdataview) SetSource(ctx session.Context, blockId string, source []stri
 	return d.Apply(s, smartblock.NoRestrictions, smartblock.KeepInternalFlags)
 }
 
+func (d *sdataview) SetSourceInSet(ctx session.Context, source []string) (err error) {
+	s := d.NewStateCtx(ctx)
+	dv, err := getDataviewBlock(s, template.DataviewBlockId)
+	if err != nil {
+		return err
+	}
+
+	for _, view := range dv.ListViews() {
+		// TODO: GO-4189 Need to review relation lists modification in each view on source change
+		view.DefaultTemplateId = ""
+		view.DefaultObjectTypeId = ""
+		if err = dv.SetView(view.Id, *view); err != nil {
+			log.With("objectId", d.Id()).Errorf("failed to update view '%s' of set: %v", view.Id, err)
```

Comment:
maybe return with error?

btw use log fields for anything except error

Reply:
Done!